pyinteraph/path_analysis.py

```diff
@@ -204,113 +214,205 @@ def get_all_shortest_paths(graph, res_id, res_space):
+#     # Get unique nodes
+#     unique_nodes = np.unique(paths_array)
+#     unique_nodes = unique_nodes[unique_nodes != "0"]
+#     # Get node percentage
+#     node_count = np.array([(node == paths_array).sum() for node in unique_nodes])
+#     node_perc = node_count/node_count.sum()
+#     # Select common nodes
+#     common_nodes = unique_nodes[node_perc > threshold]
+#     common_nodes = list(common_nodes)
+#     return common_nodes
+
+# def get_common_edges(graph, threshold):
+#     """Takes in a graph where the edge weights are the count of the edge
+#     and returns a list of edges which are larger than the provided 
+#     threshold.
+#     """
+
+#     # Initialize arrays of edges and their count
+#     edges = np.array(graph.edges)
+#     counts = np.array([graph[e[0]][e[1]]["weight"] for e in edges])
+#     # Calculate percentage occurance for each path
+#     perc = counts/counts.sum()
+#     # Choose paths larger than the threshold
+#     common_edges = edges[perc > threshold]
+#     # Convert array to list of tuples
+#     common_edges = list(map(tuple, common_edges))
+#     return common_edges
+
+def plot_graph(fname, graph, hub_num):
+    """Takes in a graph and saves a png of the plot. Also takes in a hub
+    cutoff value. Nodes with a larger number of edges than the cutoff
+    are highlighted.
+    """
+    # Get attributes
+    weights = [d["e_weight"] for u, v, d in graph.edges(data=True)]
+    nodes = np.array([n for n, d in graph.degree(graph.nodes())])
+    degrees = np.array([d for n, d in graph.degree(graph.nodes())])
+    selection = degrees >= hub_num
+    hubs = nodes[selection]
+    hubs_deg = degrees[selection]
+    non_hubs = nodes[np.logical_not(selection)]
+    non_hubs_deg = degrees[np.logical_not(selection)]
+    unique_hubs = len(np.unique(hubs_deg))
+    # Get positions. Larger k values make the nodes spread out more
+    pos = nx.spring_layout(graph, k=0.2, iterations=30)
+    # Get cmaps
+    #gray_c = sns.color_palette("gray", max(degrees) - hub_num + 1)
+    #cmap_n = LinearSegmentedColormap.from_list('gray_c', gray_c, len(gray_c))
+        # Gray scale
+    gray_c = [(0.7, 0.7, 0.7), (0.3, 0.3, 0.3)]
+    cmap_n = LinearSegmentedColormap.from_list('gray_c', gray_c, N=unique_hubs)
+    # Rocket_r
+    cb = sns.color_palette("rocket_r")
+    cmap_e = LinearSegmentedColormap.from_list('cb', cb , N=100)
+    # Remove border
+    fig, ax = plt.subplots()
+    ax.axis('off')
+    # Draw non hubs
+    nx.draw_networkx_nodes(graph, pos, node_list = list(non_hubs),
+                                       node_size = 900,
+                                       node_color = 'white',
+                                       edgecolors = 'black',
+                                       label = list(non_hubs_deg))
+    # Draw hubs
+    if unique_hubs == 1:
+        # COlor the unique hub gray
+        nx.draw_networkx_nodes(graph, pos, nodelist = list(hubs),
```

Comment:
If you are putting each argument on a separate line, put every argument on a separate line (also valid for the calls below).